src/components/navbar/index.js

```diff
@@ -0,0 +1,101 @@
+import './_navbar.scss';
+import React from 'react';
+import {connect} from 'react-redux';
+import {Redirect, Link} from 'react-router-dom';
+
+import Icon from '../icon-component';
+import Avatar from '../avatar';
+import {tokenSet} from '../../action/auth-actions.js';
+import * as util from '../../lib/utilities.js';
+import * as authActions from '../../action/auth-actions.js';
+import {profileFetchRequest} from '../../action/profile-actions.js';
+
+let NavLink = (props) => (
+  <li className={util.classToggler({selected: props.url === `/${props.route}` })} >
+    <Link to={`/${props.route}`}>
+      {props.route}
+    </Link>
+  </li>
+)
+
+class Navbar extends React.Component {
+  constructor(props){
+    super(props);
+    this.validateRoute = this.validateRoute.bind(this);
+    this.handleLogout = this.handleLogout.bind(this);
+  }
+
+  componentDidMount(){
+    this.validateRoute(this.props);
+  }
+
+  validateRoute(props){
+    let {match, history} = props;
+    let token = util.readCookie('Parkify-Token');
+
+    if(!token){
+      return history.replace('/welcome/signup');
+    }
+
+    this.props.tokenSet(token);
+    this.props.profileFetch();
+    .catch(() => {
+      console.log('PROFILE FETCH ERROR: user does not have a profile');
+      if(!match.url.startsWith('/settings')){
+        return history.replace('/settings');
+      }
+    })
+  }
+
+  handleLogout(){
+    this.props.logout();
+    this.props.history.push('/welcome/login');
+  }
+
+  render(){
+    let {url} = this.props.match
+    return (
+      <header className='navbar'>
```

Comment:
In the example from class, our header was included in our NavBar. In our case, the NavBar contains only the part that will be included in the modal opened when the user presses the menu button (i.e. the nav). We can create a separate component for NavMenu which includes the NavBar wrapped in a modal with possibly a heading.